controllers/keystoneapi_controller.go

```diff
@@ -35,6 +35,8 @@ import (
 	labels "github.com/openstack-k8s-operators/lib-common/modules/common/labels"
 	oko_secret "github.com/openstack-k8s-operators/lib-common/modules/common/secret"
 	util "github.com/openstack-k8s-operators/lib-common/modules/common/util"
+
+	//TODO1 how to import locally ?
```

Comment:
you have to update the go.mod to reference your local working dir of the db module. check the comments in https://github.com/openstack-k8s-operators/keystone-operator/blob/master/go.mod#L103-L107

Reply:
actually, I solved that (I should remove that for the time being) 
I'm using a easier  way I think, I run "go mod vendor" , then all the dependencies are downloaded to the vendor dir in the current repo main dir , under it I can manually change any file and go makes imports the vendor files  before the regular imports. 
i.e for lib-common changes I work on : 
keystone_operator_pkomarov/vendor/github.com/openstack-k8s-operators/lib-common/modules/database/funcs.go

Reply:
really not sure how that is supposed to work , what should be the flow ? 

I've tried these but : how do I know the dependency is switched? when I control click on the lib-common library after all these I still jump to ~user/go/pkg/mod ... not my working lib-common dir

- if I edit 1) and do mod tidy , it gives me an error: 


`go: errors parsing go.mod:
/home/r/my_workspace/keystone_operator_pkomarov/go.mod:92: replacement module without version must be directory path (rooted or starting with ./ or ../)`
1)replace github.com/openstack-k8s-operators/lib-common/modules/database v0.0.0-20220816094529-135dc67c2cdf => https://github.com/pinikomarov/lib-common add_database_finalizers

- if I edit 2) and go mod tidy it errors on : 
 ` (replaced by /home/r/my_workspace/lib-common_pkomarov): reading ../lib-common_pkomarov/go.mod: open /home/r/my_workspace/lib-common_pkomarov/go.mod: no such file or directory`
2)replace github.com/openstack-k8s-operators/lib-common/modules/database v0.0.0-20220816094529-135dc67c2cdf => /home/r/my_workspace/lib-common_pkomarov

- if I do edit 3),4) and mod tidy it errors out on : 
`go: conflicting replacements for github.com/openstack-k8s-operators/lib-common:
        github.com/pinikomarov/lib-common@v0.0.0-20220610121238-abedf5879ca4
        ../lib-common_pkomarov`
If I run mod tidy after just 3) then just 4) it executes ok , but on control click on a lib-common funcI still see I'm being taken to ~user/go/mod/..path to lib-common and not the ../lib-common_pkomarov I'm working on 
3)replace github.com/openstack-k8s-operators/lib-common => github.com/pinikomarov/lib-common v0.0.0-20220610121238-abedf5879ca4

4)replace github.com/openstack-k8s-operators/lib-common => ../lib-common_pkomarov

What should I do ? 
In the meantime I'm using the `go mod vendor` and working on the lib-common inside the vendor dir..
also not sure how the replace should work when I submit a PR ,I should remove the replace directive from go.mod, right ?

Reply:
works thanks, 
and how will it work on the gates? 
don't the gates have the regular (master) require libs ?

Reply:
you can not/should not push a PR with replacements to your private repo. to run tests/build image you have to use replacements to your private repose, can not use local replacements for that. to get changes to different repos, the PRs need to land first on the one it depends on. hope that answers the question.